ui/app/src/context/DataSourceRegistry.tsx

```diff
@@ -30,22 +31,25 @@ export interface DataSourceRegistryProps {
  */
 export function DataSourceRegistry(props: DataSourceRegistryProps) {
   const { children } = props;
+  const { exceptionSnackbar } = useSnackbar();
+  const { data, isLoading } = useGlobalDatasourceQuery({
```

Comment:
I mentioned this in the meeting this morning, but do you think we should change this up to wait until `isLoading` is `false` before we start rendering children? Feels like it might simplify things a bit since we probably don't want panels calling `getDataSources` before loading has completed.

Reply:
I'm a bit partial here.
Yes it totally makes sense that the panels panels shouldn't run the query if the datasource doesn't load. But it would still be nice to load the dashboard in order to edit it (when it will be possible).
In Grafana you have a load step before accessing the dashboard, and if this step doesn't work, then it's impossible to edit the dashboard. So I would like to avoid it.

Reply:
I have the feeling it will require a refactoring of how we load / register the datasources (see my other comment about how we could manage it)